src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Unit.Tests/JsonSourceGeneratorDiagnosticsTests.cs

```diff
@@ -133,20 +124,23 @@ public void NameClashSourceGeneration()
             JsonSourceGenerator generator = new JsonSourceGenerator();
             CompilationHelper.RunGenerators(compilation, out var generatorDiags, generator);
 
-            string[] expectedWarningDiagnostics = new string[] { "There are multiple types named Location. Source was generated for the first one detected. Use 'JsonSerializableAttribute.TypeInfoPropertyName' to resolve this collision." };
+            ValueTuple<TextSpan, string>[] expectedWarningDiagnostics = new ValueTuple<TextSpan, string>[]
+            {
+                (new TextSpan(303, 45), "There are multiple types named Location. Source was generated for the first one detected. Use 'JsonSerializableAttribute.TypeInfoPropertyName' to resolve this collision.")
```

Comment:
Is there a concept here of "informational" vs warning vs error?

Also can we have a test with an actual warning\error with line number?

Reply:
Yeah there are "Info", "Warning", and "Error" diagnostic severities. We only emit warning & error severities which are sufficiently tested via `Microsoft.CodeAnalysis.Location` comparisons.